# Heroes Of Pymoli Data Analysis

-----

In [2]:
# Dependencies and Setup
import pandas as pd
import numpy as np

# File to Load
file_to_load = "../../RICH201901DATA3/04-Pandas/Pandas Homework/Instructions/HeroesOfPymoli/Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_df = pd.read_csv(file_to_load)
purchase_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

## Player Count

In [3]:
#Number of Unique Players
Total_count=len(purchase_df["SN"].unique())
Total_chart = [{"Total Players": Total_count}]
Total = pd.DataFrame(Total_chart)
Total

Total Players
0            576

## **Purchasing Analysis (Total)**

In [4]:
Unique=purchase_df["Item ID"].nunique()
Average=purchase_df["Price"].mean()
Count= purchase_df["Item Name"].count()
TR= Average*Count

In [5]:
#Organize into a DF
Totals=[{"Number of Unique Items": Unique,
        "Average Price": Average,
        "Number of Purchases": Count,
        "Total Revenue": TR}]
Summary= pd.DataFrame(Totals)
Summary_table=Summary[["Number of Unique Items","Average Price","Number of Purchases","Total Revenue"]]
Summary_table

Number of Unique Items  Average Price  Number of Purchases  Total Revenue
0                     183       3.050987                  780        2379.77

## Gender Demographics

In [6]:
#Group by Gender
gender_groups=purchase_df.groupby("Gender")

#Total Count of Each Gender
gender_count = gender_groups['SN'].nunique()
GenderSum= pd.DataFrame(gender_count)

#Percent of Total Players
Percentage=(GenderSum["SN"]/Total_count)*100
GenderSum["Percentage"]=Percentage

#Rename
Sum_table=GenderSum.rename(columns={"SN":"Total Count","Percentage": "Percentage of Players"})

#Print DF
Sum_table


Total Count  Percentage of Players
Gender                                                   
Female                          81              14.062500
Male                           484              84.027778
Other / Non-Disclosed           11               1.909722


## Purchasing Analysis (Gender)

In [7]:
Gender=purchase_df.groupby("Gender")["Price"].agg('sum')
Total= purchase_df.groupby("Gender").nunique()
TotalCount=Total["Purchase ID"]

#Merge the two DataFrames
Df=pd.DataFrame(Gender)
Dftwo=pd.DataFrame(TotalCount)
merge_table= pd.merge(Df, Dftwo, on="Gender")

#Add Third Column
AveragePP=(merge_table["Price"]/TotalCount)
merge_table["Average Purchase Price"]=AveragePP

#Add Last Column
AverageTotal=(merge_table["Price"]/gender_count)
merge_table["Average Total Purchase per Person"]=AverageTotal
merge_table

#Rename
New_table=merge_table.rename(columns={"Price":"Total Purchase Value","Purchase ID": "Purchase Count"})
New_table

#Rearrange
Final_table= New_table[["Purchase Count", "Average Purchase Price", "Total Purchase Value", "Average Total Purchase per Person"]]
Final_table

Purchase Count  Average Purchase Price  \
Gender                                                          
Female                            113                3.203009   
Male                              652                3.017853   
Other / Non-Disclosed              15                3.346000   

                       Total Purchase Value  Average Total Purchase per Person  
Gender                                                                          
Female                               361.94                           4.468395  
Male                                1967.64                           4.065372  
Other / Non-Disclosed                 50.19                           4.562727

## Age Demographics

In [8]:
#Age Bins
bins=[0,9,14,19,24,29,34,39,100]
group_names=["<10","10-14", "15-19","20-24","25-29","30-34", "35-39","40+"]
purchase_df["Age Demographics"] = pd.cut(purchase_df["Age"],bins, labels=group_names)
Updated=purchase_df

#Group by Age Bins
AgeBins= Updated.groupby("Age Demographics")["SN"].nunique()
Age=pd.DataFrame(AgeBins)

#Add column to Grouby Df
PercentofP=(Age["SN"]/Total_count)*100
Age["Percent of Players"]=PercentofP

#Rename
Age_table=Age.rename(columns={"SN":"Total Count"})
Age_table


Total Count  Percent of Players
Age Demographics                                 
<10                        17            2.951389
10-14                      22            3.819444
15-19                     107           18.576389
20-24                     258           44.791667
25-29                      77           13.368056
30-34                      52            9.027778
35-39                      31            5.381944
40+                        12            2.083333

## Purchasing Analysis (Age)

In [9]:
#Group by Age
Agegroup=Updated.groupby("Age Demographics")

#Purchasing Analysis by bins
TPC= Agegroup["Price"].count()
TPV= Agegroup["Price"].sum()
APP=TPV/TPC
ATPPP= TPV/Agegroup["SN"].nunique()

final_summary=pd.DataFrame({"Purchase Count":TPC,
                    "Average Purchase Price": APP,
                    "Total Purchase Value": TPV,
                    "Avg Total Purchase Person": ATPPP})

final_summary



Purchase Count  Average Purchase Price  \
Age Demographics                                           
<10                           23                3.353478   
10-14                         28                2.956429   
15-19                        136                3.035956   
20-24                        365                3.052219   
25-29                        101                2.900990   
30-34                         73                2.931507   
35-39                         41                3.601707   
40+                           13                2.941538   

                  Total Purchase Value  Avg Total Purchase Person  
Age Demographics                                                   
<10                              77.13                   4.537059  
10-14                            82.78                   3.762727  
15-19                           412.89                   3.858785  
20-24                          1114.06                   4.318062  
25-29                           293.00                   3.805195  
30-34                           214.00                   4.115385  
35-39                           147.67                   4.763548  
40+                              38.24                   3.186667

## Top Spenders

In [10]:
TopSpenders=purchase_df.groupby("SN")

PC= TopSpenders["Purchase ID"].count()
TotalPV= TopSpenders["Price"].sum()
AveragepriceP=TotalPV/PC

Top_Spenders=pd.DataFrame({"Purchase Count":PC,
                    "Average Purchase Price": AveragepriceP,
                    "Total Purchase Value": TotalPV})
#Top_Spenders["Total Purchase Value"]=Top_Spenders["Total Purchase Value"].map("${:.2f}".format)
Top_Spenders.sort_values("Total Purchase Value", ascending=False).head()


Purchase Count  Average Purchase Price  Total Purchase Value
SN                                                                       
Lisosia93                 5                3.792000                 18.96
Idastidru52               4                3.862500                 15.45
Chamjask73                3                4.610000                 13.83
Iral74                    4                3.405000                 13.62
Iskadarya95               3                4.366667                 13.10

## Most Popular Items

In [11]:
#Group by Item ID/Item Name and find variables
Mostpop=purchase_df.groupby(["Item ID", "Item Name"])
PurchC= Mostpop["Purchase ID"].count()
ItemP=Mostpop["Price"].unique()
TPValue=Mostpop["Price"].sum()

#Create A DF
Most_popular=pd.DataFrame({"Purchase Count":PurchC,
                    "Item Price": ItemP,
                    "Total Purchase Value": TPValue})

Most_popular.sort_values("Purchase Count", ascending=False).head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm     [4.23]   
145     Fiery Glass Crusader                             [4.58]   
108     Extraction, Quickblade Of Trembling Hands        [3.53]   
82      Nirvana                                           [4.9]   
19      Pursuit, Cudgel of Necromancy                    [1.02]   

                                                      Total Purchase Value  
Item ID Item Name                                                           
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
145     Fiery Glass Crusader                                         41.22  
108     Extraction, Quickblade Of Trembling Hands                    31.77  
82      Nirvana                                                      44.10  
19      Pursuit, Cudgel of Necromancy                                 8.16

## Most Profitable Items

* Sort the above table by total purchase value in descending order


In [16]:
Most_popular.sort_values("Total Purchase Value", ascending=False).head()

Purchase Count Item Price  \
Item ID Item Name                                                      
63      Stormfury Mace                                  2     [4.99]   
29      Chaos, Ender of the End                         5     [1.98]   
173     Stormfury Longsword                             2     [4.93]   
1       Crucifer                                        3     [3.26]   
38      The Void, Vengeance of Dark Magic               4     [2.37]   

                                          Total Purchase Value  
Item ID Item Name                                               
63      Stormfury Mace                                   $9.98  
29      Chaos, Ender of the End                          $9.90  
173     Stormfury Longsword                              $9.86  
1       Crucifer                                         $9.78  
38      The Void, Vengeance of Dark Magic                $9.48

# Key Takeaways:

* 84 % of players are male

* The largest age demopgraphic bucket is age 20-24

* Players have spent the most on Oathbrajer, Last Hope of the Breaking Storm
